lib/google-analytics.js

```diff
@@ -122,17 +127,41 @@ exports.track = function(title, data) {
   exports.send('event', params);
 };
 
-exports.listen = function(app) {
-  if (!app.isFeatureEnabled('google-analytics')) {
-    debug('not enabled');
+exports.unlisten = function() {
+  if (!exports.setupDone) {
     return;
   }
+  if (exports.app) {
+    if (exports.app.user) {
+      debug('removing listener to user');
+      exports.app.user.off('sync', exports.setUser);
+    }
+
+    if (exports.app.router) {
+      debug('removing listener to router');
+      exports.app.router.off('page', exports.update);
+    }
+  }
+  exports.setupDone = false;
+};
 
+exports.listen = function(app) {
   exports.app = app;
 
+  if (!app.isFeatureEnabled('googleAnalytics')) {
+    exports.unlisten();
+    debug('google analytics is disabled');
+    return;
+  }
+
   if (app.user) {
     debug('adding listener to user');
-    app.user.on('sync', exports.setUser.bind(null, app.user));
+    // if user already fetched, set user now
+    if (app.user.fetched) {
+      exports.setUser(app.user);
```

Comment:
You are passing "app.user" as an argument to setUser, but setUser no longer takes an argument above. Do you still want it to take an argument?